ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/ParallelIndexerFactory.scala

```diff
@@ -110,13 +111,18 @@ object ParallelIndexerFactory {
                 dbDispatcher = dbDispatcher,
                 materializer = mat,
               )
-            ).flatMap { handle =>
+            ).transform {
```

Comment:
This looks like [Future.andThen](https://www.scala-lang.org/api/current/scala/concurrent/Future.html#andThen[U](pf:PartialFunction[scala.util.Try[T],U])(implicitexecutor:scala.concurrent.ExecutionContext):scala.concurrent.Future[T]):

```suggestion
            ).andThen {
              case Success(handle) => killSwitchPromise.success(handle.killSwitch)
              case Failure(ex) => killSwitchPromise.failure(ex)
            }.flatMap(_.completed)
```

Reply:
nice!